dp/LCS_string.py

```diff
@@ -0,0 +1,59 @@
+# def LCSS(str1: str, str2: str, strlen1: int, strlen2: int):
+# 	ret = []
+
+# 	if str1[-1] == str2[-1]:
+# 		return LCSS(str1[:strlen1-1], str2[:strlen2-1], strlen1-1, strlen2-1) + str1[-1]
+
+# 	max(len(LCSS(str1[:strlen-1], str2)), len(LCSS(str1, str2[:strlen2-1]))), 
+
+# 	return ""
+
+# def LCS(A: str, lenA: int, B: str, lenB: int):
+# 	if lenA <= 0 or lenB <= 0:
+# 		return 0
+# 	print("A: " + A[0:lenA])
+# 	if A[-1] == B[-1]:
+# 		return LCS(A, lenA-1, B, lenB-1) + 1
+	
+# 	return max(LCS(A, lenA-1, B, lenB), LCS(A, lenA, B, lenB-1))
+
+def LCS(A: str, B: str):
```

Comment:
Run this code through pylint, and see what it says to improve it.

You dont have to take all its suggestions, but it is good to see what codestyle looks like.

for instance, it will complain profusely about your choice of variable names, A and B.   Generally that is a good thing, but here, maybe we dont need it, so we can ignore that.